support-frontend/app/services/ProductCatalogService.scala

```diff
@@ -0,0 +1,58 @@
+package services
+
+import com.gu.okhttp.RequestRunners.FutureHttpClient
+import com.gu.rest.WebServiceHelper
+import io.circe.{Decoder, Json, JsonObject}
+import io.circe.generic.semiauto.deriveDecoder
+import org.apache.pekko.actor.ActorSystem
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+case class ProductCatalogServiceError(message: String) extends Throwable
+object ProductCatalogServiceError {
+  implicit val decoder: Decoder[ProductCatalogServiceError] = deriveDecoder
+}
+
+trait ProductCatalogService extends WebServiceHelper[ProductCatalogServiceError] {
+  val client: FutureHttpClient
+  override val wsUrl: String
+  override val httpClient: FutureHttpClient = client
+
+  def get(): Future[JsonObject] = {
+    get[JsonObject](endpoint = "product-catalog.json")
+  }
+}
+
+class ProdProductCatalogService(val client: FutureHttpClient) extends ProductCatalogService {
+  val wsUrl: String = "https://product-catalog.guardianapis.com"
+}
+
+class CodeProductCatalogService(val client: FutureHttpClient) extends ProductCatalogService {
+  val wsUrl: String = "https://product-catalog.code.dev-guardianapis.com"
+}
+
+class CachedProductCatalogService(system: ActorSystem, productCatalogService: ProductCatalogService)(implicit
+    ec: ExecutionContext,
+) {
+  private val json = new AtomicReference[JsonObject](JsonObject())
```

Comment:
I'm using a JsonObject as I am hoping to keep this data only relevant to the client, which [has its own parser](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/pages/%5BcountryGroupId%5D/checkout.tsx#L138-L150).